### DISCUSSION TITLE
Why isn't the new commit assistance available in Copilot?

### DISCUSSION THREAD

**Discussion by rfalanga** [community/community#120505] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have purchased GitHub Copilot for an individual. Recently I was editing some C# code, then I went to commit my changes locally. I looked for the GitHub Copilot assistance button I've seen demonstrated to help generate a commit statement. It wasn't there.

Why not?

**Comment by Youssef-BS** [comment]
Hello , github copilot is continuously being updated, and some features might not be available across all languages or environments make sure that the feature you're looking for is indeed supported for C# code.

**Reply by rfalanga** [reply]
I believe it is. 

One thing, I was working on a repo that is in Azure DevOps Services. I would think that would still work, but could that be the difference? Is this feature of GitHub Copilot only for repositories on GitHub?

**Reply by ayim** [reply]
GitHub Copilot works on code even if it's not attached to a repository - whether local or remote

**Comment by ayim** [comment]
Which IDE are you using?

**Reply by rfalanga** [reply]
Visual Studio 2022 Community edition

**Reply by ayim** [reply]
Which version of the IDE are you using? If you're on the latest, the assistance should be available

**Reply by rfalanga** [reply]
I'm using version 17.9.6 of Visual Studio 2022 Community Edition.

Could it be the "size" of the commit? I made only one small change, because I was curious to know what Copilot might do. Could that have been too small?

**Reply by ayim** [reply]
It should show whether a commit exists or not.
Is "Git preview features" checked for you in Tools > Options > GitHub > Copilot?
![image](https://github.com/community/community/assets/10951869/42dc4e18-be29-4ed1-8e01-e1691eedef68)

**Reply by rfalanga** [reply]
Nope. My `Tools > Options > GitHub > Copilot` looks like this:

![image](https://github.com/community/community/assets/11872620/9a025fa2-871e-475d-bdfc-2360a1d1b354)

**Reply by ayim** [reply]
Have you got this extension installed?
https://marketplace.visualstudio.com/items?itemName=VisualStudioExptTeam.VSGitHubCopilot

**Reply by rfalanga** [reply]
@ayim no, I don't have the GitHub Copilot Chat extension installed. I wasn't even aware of it. I see that it has a free trial, so what does it cost? I've purchased a year subscription to GitHub Copilot, does that mean that I can get GitHub Copilot Chat extension for free or is that an additional expense?

**Reply by rfalanga** [reply]
I'm revisiting this because I still don't know if I need to install GitHub Copilot Chat Extension, so that I can get assistance when creating a commit in Visual Studio. Do I need GitHub Copilot Chat Extension?

**Reply by ayim** [reply]
Yes, you should have access to the Chat extension as a part of your Copilot subscription. You will need to install it - the extension provides AI commit history functionality in 17.10+ builds

**Reply by rfalanga** [reply]
Is that the `GitHub Copilot Completions` for Visual Studio?

**Reply by rfalanga** [reply]
@ayim, I've just installed the GitHub Copilot Chat Extension. Will have to test it later.

Thank you

**Reply by rfalanga** [reply]
I've installed the GitHub Copilot Chat Extension into VS 2022. Then I exited out of VS 2022 and got back into it. I made a few changes and went to perform a commit, to see if I could get commit message suggestions. I didn;'t. 😞 Here's what I now have in Tools | Options | GitHub | Copilot chat:

![image](https://github.com/community/community/assets/11872620/51608edf-f691-4ed7-b390-72dcb5f8b6f5)

Have I something more which I need to do?

**Reply by ayim** [reply]
You should be seeing the icon in the commit message box at this point.
In Visual Studio, please have the commit panel open, then click the "Report a Problem" button available in the top right of Visual Studio so that we can investigate further

**Reply by rfalanga** [reply]
OK, @ayim, I'll do that now. Thank you.

**Reply by rfalanga** [reply]
Its done.

**Comment by rfalanga** [comment]
Recently Microsoft released version 17.10.0 for Visual Studio 2022. Since installing it, I've noticed that GitHub Copilot now includes the Copilot Commit Assistant. So, I think that VS version has resolved my issue.

**Reply by ayim** [reply, marked as answer]
Great! We're continuing to work on the commit scenarios, really excited about this space
